src/julia/Integer.jl

```diff
@@ -606,7 +606,7 @@ RandomExtensions.maketype(R::AbstractAlgebra.Integers{T}, _) where {T} = T
 
 # define rand(make(ZZ, n:m))
 rand(rng::AbstractRNG,
-     sp::SamplerTrivial{<:Make2{T, Integers{T}, UnitRange{Int}}}
+     sp::SamplerTrivial{<:Make2{T, Integers{T}, <:UnitRange{<:Integer}}}
```

Comment:
@rfourquet this seems to work for me in practice, does it it seem plausible to your? Also, I wonder if it should be changed further to
```suggestion
     sp::SamplerTrivial{<:Make2{T, Integers{T}, <:AbstractRange{<:Integer}}}
```
Then I can even do stuff like `rand(ZZ, 0:2:10)`. Any downsides I am missing?

Reply:
No I don't see any dowside with this. I didn't do it when porting the `rand` code to use `RandomExtensions` to keep the change minimal, but I was definitely furstrated by the unnecessary (in my eyes) constraint (and I didn't try to extend it later as I had received push back in another occasion (#489) when trying to use `Integer` instead of `Int`, although the arguments there might not have applied here with `rand`).

Actually, I would recommand `<:AbstractArray{<:Integer}` here.

Reply:
Thanks for the feedback, @rfourquet -- and done.

Reply:
Do you see a drawback with using `AbstractArray` instead of `AbstractRange`?

Reply:
Nope, I simply misread and didn't see your earlier suggestion. I'll update to that.